cargo-miri/bin.rs

```diff
@@ -617,13 +645,16 @@ fn phase_cargo_rustc(args: env::Args) {
         }
 
         // Use our custom sysroot.
-        let sysroot =
-            env::var_os("MIRI_SYSROOT").expect("the wrapper should have set MIRI_SYSROOT");
         cmd.arg("--sysroot");
         cmd.arg(sysroot);
     } else {
         // For host crates or when we are printing, just forward everything.
         cmd.args(args);
+
+        if env::var_os("MIRI_TARGET_IS_HOST").is_some() {
+            cmd.arg("--sysroot");
+            cmd.arg(sysroot);
```

Comment:
> This sounds like it is using the host rustc sysroot when it needs the Miri sysroot.

It's strange that I explicitly set `--sysroot` to the Miri sysroot. I must miss something. :confused:

Reply:
I don't understand your goal here. Didn't we say that these crates (proc-macro tests) would *not* be interpreted with Miri, and that they would not be run at all?

Reply:
@RalfJung That's not what I was trying to do here. I updated #1675 and did that there.